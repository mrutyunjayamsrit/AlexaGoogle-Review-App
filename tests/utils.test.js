const utils = require('../utils/utils')
jest.mock('../utils/utils')
const content = './data/review.json'

describe('Tests should verify all utils functions', ()=>{

    it('Should verify getAllReviewsByDefault to get reviews', async ()=>{
        
        let reviews = {};
        const spy = jest.spyOn(utils, 'getAllReviewsByDefault');
        spy.mockImplementation(() => Promise.resolve({
            reviews: [{
                review: "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
                author: "_Reef_",
                review_source: "iTunes",
                rating: 1,
                title: "Does Jeff Bezos use this?",
                product_name: "Amazon Alexa",
                reviewed_date: "2017-12-18T22:04:04.000Z"
                }]
          })); 
        reviews = await utils.getAllReviewsByDefault();
       
        expect(reviews).toMatchSnapshot();
        spy.mockRestore();
    })

    it('Should verify getReviewsonGivenDate to get reviews', async ()=>{
        
        let reviews = content;
        let date = '2017-12-18'
        const spy = jest.spyOn(utils, 'getReviewsonGivenDate');
        spy.mockImplementation(() => Promise.resolve({
            reviews: [{
                review: "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
                author: "_Reef_",
                review_source: "iTunes",
                rating: 1,
                title: "Does Jeff Bezos use this?",
                product_name: "Amazon Alexa",
                reviewed_date: "2017-12-18T22:04:04.000Z"
                }]
          })); 
        reviews = await utils.getReviewsonGivenDate(date);
       
        expect(reviews).toMatchSnapshot(content);
        spy.mockRestore();
    })

    it('Should verify getReviewsByGivenRating to get reviews', async ()=>{
        
        let reviews = content;
        let rating = 1
        const spy = jest.spyOn(utils, 'getReviewsByGivenRating');
        spy.mockImplementation(() => Promise.resolve({
            reviews: [{
                review: "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
                author: "_Reef_",
                review_source: "iTunes",
                rating: 1,
                title: "Does Jeff Bezos use this?",
                product_name: "Amazon Alexa",
                reviewed_date: "2017-12-18T22:04:04.000Z"
                }]
          })); 
        reviews = await utils.getReviewsonGivenDate(rating);
       
        expect(reviews).toMatchSnapshot(content);
        spy.mockRestore();
    })

    it('Should verify getReviewsByGivenStore to get reviews', async ()=>{
        
        let reviews = content;
        let store = "i;Tunes"
        const spy = jest.spyOn(utils, 'getReviewsByGivenStore');
        spy.mockImplementation(() => Promise.resolve({
            reviews: [{
                review: "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
                author: "_Reef_",
                review_source: "iTunes",
                rating: 1,
                title: "Does Jeff Bezos use this?",
                product_name: "Amazon Alexa",
                reviewed_date: "2017-12-18T22:04:04.000Z"
                }]
          })); 
        reviews = await utils.getReviewsByGivenStore(store);
       
        expect(reviews).toMatchSnapshot(content);
        spy.mockRestore();
    })
})