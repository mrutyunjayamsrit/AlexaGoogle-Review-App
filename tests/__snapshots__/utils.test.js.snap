// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests should verify all utils functions Should verify getAllReviewsByDefault to get reviews 1`] = `
Object {
  "reviews": Array [
    Object {
      "author": "_Reef_",
      "product_name": "Amazon Alexa",
      "rating": 1,
      "review": "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
      "review_source": "iTunes",
      "reviewed_date": "2017-12-18T22:04:04.000Z",
      "title": "Does Jeff Bezos use this?",
    },
  ],
}
`;

exports[`Tests should verify all utils functions Should verify getAllReviewsByDefault to get reviews: ./data/review.json 1`] = `
Object {
  "reviews": Array [
    Object {
      "author": "_Reef_",
      "product_name": "Amazon Alexa",
      "rating": 1,
      "review": "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
      "review_source": "iTunes",
      "reviewed_date": "2017-12-18T22:04:04.000Z",
      "title": "Does Jeff Bezos use this?",
    },
  ],
}
`;

exports[`Tests should verify all utils functions Should verify getReviewsByGivenRating to get reviews: ./data/review.json 1`] = `undefined`;

exports[`Tests should verify all utils functions Should verify getReviewsByGivenStore to get reviews: ./data/review.json 1`] = `
Object {
  "reviews": Array [
    Object {
      "author": "_Reef_",
      "product_name": "Amazon Alexa",
      "rating": 1,
      "review": "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
      "review_source": "iTunes",
      "reviewed_date": "2017-12-18T22:04:04.000Z",
      "title": "Does Jeff Bezos use this?",
    },
  ],
}
`;

exports[`Tests should verify all utils functions Should verify getReviewsonGivenDate to get reviews 1`] = `undefined`;

exports[`Tests should verify all utils functions Should verify getReviewsonGivenDate to get reviews: ./data/review.json 1`] = `
Object {
  "reviews": Array [
    Object {
      "author": "_Reef_",
      "product_name": "Amazon Alexa",
      "rating": 1,
      "review": "No point in stating the obvious, app is garbage right now. I like Alexa but find it hard to believe Bezos uses this glitchy app.",
      "review_source": "iTunes",
      "reviewed_date": "2017-12-18T22:04:04.000Z",
      "title": "Does Jeff Bezos use this?",
    },
  ],
}
`;
